assets/js/modules/analytics/datastore/report.test.js

```diff
@@ -425,5 +419,72 @@ describe( 'modules/analytics report', () => {
 				expect( isNotGathered ).toBe( false );
 			} );
 		} );
+
+		describe( 'hasZeroData', () => {
+			it( 'should return undefined if getReport or isGatheringData is not resolved yet', async () => {
+				freezeFetch( analyticsRegexp );
+
+				const hasZeroData = registry
+					.select( MODULES_ANALYTICS )
+					.hasZeroData();
+
+				expect( hasZeroData ).toBeUndefined();
+			} );
+
+			it.each( [
+				[ 'an empty array', [ { data: { rows: [] } } ] ],
+				[ 'null', null ],
+			] )(
+				'should return TRUE if report data in isGatheringData OR isZeroReport is %s',
```

Comment:
Where in this test is `isGatheringData` vs `isZeroReport` being tested?

It looks like you're testing `isGatheringData` with the `null` report and `isZeroReport` with the empty array. Is that correct?

If so, these should be split up into two separate tests with the relevant result for each one.

Again, see my above comment about DRY in tests; it's tempting to use this to test both options with fewer lines of code, but minimising LOC in a test file is counter-intuitive, because when this test fails it won't be as clear why/which is failing. Doubly-so because the test spec names which selector is being tested in opposite order to the `it.each` array, I think.

Reply:
Agreed 👍

Reply:
>It looks like you're testing isGatheringData with the null report and isZeroReport with the empty array. Is that correct?
Yes.